src/lib.rs

```diff
@@ -245,10 +245,19 @@ impl Executor {
             let id = ID.fetch_add(1, Ordering::Relaxed);
 
             // Spawn the new thread.
-            thread::Builder::new()
+            if let Err(e) = thread::Builder::new()
                 .name(format!("blocking-{}", id))
                 .spawn(move || self.main_loop())
-                .unwrap();
+            {
+                // We were unable to spawn the thread, so we need to undo the state changes.
+                log::error!("Failed to spawn a blocking thread: {}", e);
+                inner.idle_count -= 1;
+                inner.thread_count -= 1;
+
+                // The current number of threads is likely to be the system's upper limit, so update
+                // thread_limit accordingly.
```

Comment:
(I wonder what happens if the thread creation fails when the thread_count is 0...)

Reply:
Right now, the current strategy is to just not run whatever runnables are in the queue, and wait for `grow_pool` to be invoked again on the off chance that whatever resource exhaustion prevented us from spawning a thread in the first place isn't happening anymore. However, this could lead to deadlocks on constrained systems. Panicking is also an option, but that feels like it detracts from the entire point.

Reply:
when is `thread_limit` increased again?

Reply:
`thread_limit` is first set at the start of the program to either a default of 500 or to a user-defined value. Then, if there is an error while spawning a thread, `thread_limit` is set to the current thread count. It is never increased.

Reply:
Can we ensure that `thread_limit` never drops to 0?

Reply:
I've changed it so that, if it is about to be set to zero, it is set to one instead. I still don't think panicking is appropriate here.

Reply:
ok.